[brennalaurentan] iP

brennalaurentan opened the pull request:
### Tam the Task Manager 😄 

> "A goal without a plan is just a wish." -Antoine de Saint-Expuéry ([source](https://www.goodreads.com/quotes/87476-a-goal-without-a-plan-is-just-a-wish))

Calling all CLI lovers!
The newest text-based task manager is here and it's completely **free!**

Almost like plug-and-play, here's how you can get started:

1. Download the jar file from this repository
2. Run the jar file
3. Start using `Tam`!

Some features coming up:
- [ ] GUI

If you have any of the following:
- Feedback on how the app can better serve your needs
- Encountered some bugs

Feel free to leave your comments on this repository!

```
// have a nice day!
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

brennalaurentan committed:
Removed Duke class

brennalaurentan committed:
Added Tam class with greet and exit texts

brennalaurentan committed:
Added Echo feature

brennalaurentan committed:
Added add task and list features

brennalaurentan committed:
Added Task class and Mark/Unmark Task feature

brennalaurentan committed:
Added Todo, Deadline, Event classes

brennalaurentan committed:
Added automated testing using I/O redirection

brennalaurentan committed:
Added basic error handling for incorrect inputs for all Todo, Deadline, Event task types (missing parameters, empty parameters)

brennalaurentan committed:
Added delete task feature with error handling

brennalaurentan committed:
Added enumeration usage with Commands enum

brennalaurentan committed:
Updated Help feature with new Delete command

brennalaurentan committed:
Add formatting between String and Task form for all Task subclasses, for saving feature

brennalaurentan committed:
Add FileRetriever class for list saving and retrieving between sessions

brennalaurentan committed:
Add list retrieving on startup and saving after changes made

brennalaurentan committed:
Sample saved list

brennalaurentan committed:
Add LocalDate formatting for saving due, from and to dates for Deadline and Event tasks

brennalaurentan committed:
Refactor code into Ui, Storage, Parser, TaskList

brennalaurentan committed:
Add delete all tasks feature

AndrewOng2066 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,218 @@
+                        eventFromToArray = eventDescriptionFromToArray[1].trim().split(" /to ");
+                        eventFrom = eventFromToArray[0];
+                        eventTo = eventFromToArray[1];
+                    }
```
Can consider leaving a blank line after this line of code to enhance the readability.

"Logical units within a block should be separated by one blank line." - Java coding standard website.

AndrewOng2066 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,218 @@
+                        System.out.print(dividerText);
+                        Storage.saveTaskList(existingTaskList);
+                        return 1;
+                    }
```
Can consider leaving a blank line after the line to enhance the readability.

"Logical units within a block should be separated by one blank line." - Java coding standard website.

AndrewOng2066 reviewed (commented):
Well done! Couldn't find any violation regarding Java coding standard. However, you may consider separating some code by a blank line to enhance the readability.

brennalaurentan committed:
Organise into packages

brennalaurentan committed:
Merge branch 'add-gradle-support'

brennalaurentan committed:
Update Storage.java

brennalaurentan committed:
Add JUnit testing

brennalaurentan committed:
Update build.gradle with JAR details

brennalaurentan committed:
Add JavaDoc comments to all public methods

brennalaurentan committed:
Refactor code to adhere to coding standards

brennalaurentan committed:
Add Find feature

brennalaurentan committed:
Remove unused imports

brennalaurentan committed:
Merge branch 'branch-A-CodingStandard'

brennalaurentan committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/parser/Parser.java

brennalaurentan committed:
Refactor code to adhere to coding standards

timothysashimi commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,95 @@
+     *
+     * @param saveString Deadline task in string format
+     * @return Deadline task as a Deadline object
+     */
+    public Task convertSaveToTask(String saveString) {
+        //T|1|READ BOOK|dueDate
```
Maybe the method name can be phrased better? convert save to task sounds a bit weird haha

timothysashimi commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,95 @@
+     * Converts a Deadline task from Deadline object to string (for saving)
+     *
+     * @return Deadline task as a string
+     */
+    public String convertTaskToSave() {
+        String taskCode = this.getTaskCode();
```
this too! can it be phrased in another way?

timothysashimi commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,188 @@
+package tasks;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class TaskList {
+
```
Love how neat your code is!

timothysashimi reviewed (commented):
LGTM!!

timothysashimi commented:
LGTM!

brennalaurentan committed:
Add comments for each of the classes

brennalaurentan committed:
Implement Checkstyle with Gradle

brennalaurentan committed:
Fix bug causing error in testing

brennalaurentan committed:
Add GUI using JavaFX

brennalaurentan committed:
Reconfigure to create new jar file

brennalaurentan committed:
Update 'magic numbers' with named constant

Currently returning 0 or 1 from the parseAndExecuteCommand() method
to represent a terminating or non-terminating command respectively.

Let's use named constant integers to represent the 0 and 1 instead.

brennalaurentan committed:
Rewrite code to avoid complicated expressions

brennalaurentan committed:
Update Commands enumeration with specialized help messages

Instructions provided on how to use each command (shown
when user types the 'help' command)  is currently statically
typed and printed in the parseAndExecuteCommand()
method's switch.

Placing the instructions for each of the possible commands in the
Commands enumeration makes it easier to maintain and add to.

Let's make sure to update the enumeration each time we add
a new command instead of cluttering the parseAndExecuteCommand()
method

brennalaurentan committed:
Update parseAndExecuteCommand() to pull help messages from enumeration

brennalaurentan committed:
Implement java assert to check actual against expected values

brennalaurentan committed:
Break down logic in parseAndExecuteCommand() into multiple different
methods

The logic for the parseAndExecuteCommand() method covers a
number of purposes - parsing command input, validating command input,
and executing the command.

This makes debugging especially painful as the code is messy.

Following SLAP and to reduce too long methods, let's break large
methods into smaller ones which each serve a single purpose

brennalaurentan committed:
Merge pull request #1 from brennalaurentan/branch-A-Assertions

Implement java assert to check actual against expected values

brennalaurentan committed:
Revert "Implement java assert to check actual against expected values"

brennalaurentan committed:
Merge pull request #3 from brennalaurentan/revert-1-branch-A-Assertions

Revert "Implement java assert to check actual against expected values"

brennalaurentan committed:
Merge pull request #2 from brennalaurentan/branch-A-CodeQuality

Enhance code quality by following coding standards

brennalaurentan committed:
Merge pull request #4 from brennalaurentan/master

Implement java assertions to check actual against expected values

brennalaurentan committed:
Merge pull request #5 from brennalaurentan/branch-A-Assertions

Merge pull request #4 from brennalaurentan/master

brennalaurentan committed:
Implement comparing actual and expected values using java assert

brennalaurentan committed:
Merge pull request #6 from brennalaurentan/branch-A-Assertions

Implement comparing actual and expected values using java assert

brennalaurentan committed:
Add 'help' feature using streams to iterate over enumeration of commands

Help feature is implemented by printing a list of statically typed
instructions for each of the commands listed in the Commands enumeration

Each time a new command type is added, it is added to the enumeration
along with its 'help message' (instructions for how to use the command)

Compilation of the help message requires you to obtain the string by
calling command.getHelpMessage() for each of the commands

Using streams to iterate over the enumeration to generate the list is
much more efficient.

Let's use streams where possible to reduce the amount of hard coding.

brennalaurentan committed:
Fix bug where commands typed in caps did not behave as expected

Todo, Deadline and Event commands typed in capital letters
(TODO, DEADLINE, EVENT) ended up getting appended to the task
description

Use regex to bypass case-sensitivity when identifying keywords

Let's make commands not case-sensitive so users can have more
flexibility when using Tam.

brennalaurentan committed:
Add .yml file to set up Continuous Integration (CI)

brennalaurentan committed:
Update code to comply with CheckStyle requirements

brennalaurentan committed:
Fix Java CI workflow error

brennalaurentan committed:
Update code to comply with CheckStyle requirements

brennalaurentan committed:
Create Help.md

brennalaurentan committed:
Rename Help.md to HELP.md

brennalaurentan committed:
Create HELP.md

brennalaurentan committed:
Delete HELP.md

brennalaurentan committed:
Create HELP.md

brennalaurentan committed:
Delete HELP.md

brennalaurentan committed:
Update HELP.md

brennalaurentan committed:
Update HELP.md with list of commands

brennalaurentan committed:
Implement hyperlink to external help page for C-Help extension

brennalaurentan committed:
Add necessary JavaDoc for all public methods and classes

brennalaurentan committed:
Add JUnit tests

brennalaurentan committed:
Update JUnit tests

brennalaurentan committed:
Add JUnit tests to test possible inputs more comprehensively

brennalaurentan committed:
Fix error in Validator logic for Deadline commands

brennalaurentan committed:
Upload screenshot of ui

brennalaurentan committed:
Update README.md

brennalaurentan committed:
Update README.md

brennalaurentan committed:
Update README.md

brennalaurentan committed:
Update README.md

brennalaurentan committed:
Update name of jar file to be generated

brennalaurentan committed:
Delete README.md
